README.md

```diff
@@ -217,13 +217,17 @@ ios/tmp.xcconfig
 4. go to project settings
 5. apply config to your configurations
    ![img](./readme-pics/3.ios_apply_config.png)
-6. create new build phase for the scheme which will generate "tmp.xcconfig" before each build exposing values to Build Settings and Info.plist (this snippet has to be placed after "cp ... ${PROJECT_DIR}/../.env" if [approach explained below](#ios-multi-scheme) is used)
+6. Go to _Edit scheme..._ -> _Build_ -> _Pre-actions_, click _+_ and select _New Run Script Action_. Paste below code which will generate "tmp.xcconfig" before each build exposing values to Build Settings and Info.plist. Make sure to select your target under _Provide build settings from_, so `$SRCROOT` environment variables is available to the script. (Note that this snippet has to be placed after "cp ... \${PROJECT_DIR}/../.env" if [approach explained below](#ios-multi-scheme) is used).
 
-```
-"${SRCROOT}/../node_modules/react-native-config/ios/ReactNativeConfig/BuildXCConfig.rb" "${SRCROOT}/.." "${SRCROOT}/tmp.xcconfig"
-```
+   ```
+   "${SRCROOT}/../node_modules/react-native-config/ios/ReactNativeConfig/BuildXCConfig.rb" "${SRCROOT}/.." "${SRCROOT}/tmp.xcconfig"
+   ```
+
+   ![img](./readme-pics/4.ios_pre_actions.png)
+
+   Note that you need to apply this pre-action on every scheme you use. _Build_ and _Archive_ is a bare minimum.
```

Comment:
Do you really need to set this pre-action in _Archive_ after setting it in _Build_?

I set it up in _Build_ and it works for all the other modes.

Reply:
Does it populate values in `Info.plist` if you remove `tmp.xcconfig` in the file system and then run an Archive scheme without running Build first (fresh build) or does it pick up the change on the first Archive run after you've changed value in your `.env`? 

It works only `tmp.xcconfig` was already generated, which bit me pretty hard in CI and took quite some time to figure out.

Reply:
We may have different configurations, so that's actually a question :) Would be nice if we can get it working with adding Pre-actions only in one place.

Reply:
I also set it up only in _Build_ step and didn't see any issues. I tried `Build`, `Archive`, `Clean Build Folder` - it generates the `tmp.xcconfig` file before all of these actions.

Reply:
Interesting. It does indeed re-generate the file, but it seems to happen too late. In my case I reference a variable defined in `tmp.xcconfig` (`APP_NAME`) from the `Info.plist` file. If I remove `tmp.xcconfig`, then run Archive (without it having a pre-action) and then check my `Info.plist` in the build directory (printed at the end of the build log), then `Info.plist` in that directory is missing the value. Can you confirm that it works for you in this scenario?

```
	<key>CFBundleDisplayName</key>
	<string>$(APP_NAME)</string>
```

Reply:
Yes, it works for me. I am using react-native-config to set an URL type in my Info.plist with google sign in's reversed client id set in my .env. My steps:
1. Clean Build Folder
2. I removed the `.tmp.xcconfig` file
3. Run Archive
...and all worked out perfectly well! 🎉

Reply:
So I made a fresh RN project with the react-native-config: https://github.com/devoto13/rnc-pre-action

I clone this project, `cd ios; pod install`, open project in Xcode, set my development team and bundle identifier on Signing & Capabilities and click Run. The app on my phone runs and has *Custom Name* as the name. So far so good.

Then I run Product -> Archive, poor myself a cup of coffee or two, and then check `/Users/devoto13/Library/Developer/Xcode/Archives/2020-10-28/MyApp 2020-10-28, 18.11.xcarchive/Products/Applications/MyApp.app/Info.plist` in Xcode. Bundle display name is set to *Custom Name*. Nice!

Now I remove `tmp.xcconfig` and run Product -> Archive again. Go have a good chat with a colleague and then check `/Users/devoto13/Library/Developer/Xcode/Archives/2020-10-28/MyApp 2020-10-28, 18.20.xcarchive/Products/Applications/MyApp.app/Info.plist` in Xcode. And it still has Bundle display name set to *Custom Name*.

So you're right. Maybe something is misconfigured in my real application or `fastlane` executes Archive differently than Xcode. I'm too lazy to debug now 😄 I'll remove the clause about adding to every scheme from the docs to avoid confusing people.